File: go/TennisGame3.go
package tenniskata

type tennisGame3 struct {
	p2  int
	p1  int
	p1N string
	p2N string
}

func TennisGame3(p1N string, p2N string) TennisGame {
	game := &tennisGame3{
		p1N: p1N,
		p2N: p2N}

	return game
}

func (game *tennisGame3) GetScore() string {
	var s string
	if game.p1 < 4 && game.p2 < 4 && !(game.p1+game.p2 == 6) {
		p := []string{"Love", "Fifteen", "Thirty", "Forty"}
		s = p[game.p1]
		if game.p1 == game.p2 {
			return s + "-All"
		}
		return s + "-" + p[game.p2]
	} else {
		if game.p1 == game.p2 {
			return "Deuce"
		}
		if game.p1 > game.p2 {
			s = game.p1N
		} else {
			s = game.p2N
		}
		if (game.p1-game.p2)*(game.p1-game.p2) == 1 {
			return "Advantage " + s
		}
		return "Win for " + s
	}
}

func (game *tennisGame3) WonPoint(playerName string) {
	if playerName == "player1" {
		game.p1 += 1
	} else {
		game.p2 += 1
	}
}

File: php/src/TennisGame3.php
<?php

declare(strict_types=1);

namespace TennisGame;

class TennisGame3 implements TennisGame
{
    private int $p2 = 0;

    private int $p1 = 0;

    public function __construct(
        private string $p1N,
        private string $p2N
    ) {
    }

    public function getScore(): string
    {
        if ($this->p1 < 4 && $this->p2 < 4 && ! ($this->p1 + $this->p2 === 6)) {
            $p = ['Love', 'Fifteen', 'Thirty', 'Forty'];
            $s = $p[$this->p1];
            return ($this->p1 === $this->p2) ? "{$s}-All" : "{$s}-{$p[$this->p2]}";
        }
        if ($this->p1 === $this->p2) {
            return 'Deuce';
        }
        $s = $this->p1 > $this->p2 ? $this->p1N : $this->p2N;
        return (($this->p1 - $this->p2) * ($this->p1 - $this->p2) === 1) ? "Advantage {$s}" : "Win for {$s}";
    }

    public function wonPoint(string $playerName): void
    {
        if ($playerName === 'player1') {
            $this->p1++;
        } else {
            $this->p2++;
        }
    }
}
